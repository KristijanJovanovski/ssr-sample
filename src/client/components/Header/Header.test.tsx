import * as React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { MemoryRouter } from 'react-router-dom';
import { Header } from '.';

test('should render self and sub-components', () => {
  const tree = mount(
    <MemoryRouter initialEntries={['/']}>
      <Header userName="foo" orgName="org!" />
    </MemoryRouter>
  );

  expect(toJson(tree)).toMatchSnapshot();
});
