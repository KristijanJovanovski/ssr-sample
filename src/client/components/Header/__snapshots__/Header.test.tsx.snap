// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render self and sub-components 1`] = `
.c0 {
  background: #333;
  color: #f5f5f5;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 15px 30px;
}

<MemoryRouter
  initialEntries={
    Array [
      "/",
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "b8jb8s",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "b8jb8s",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Component
      name="foo"
    >
      <styled.header>
        <header
          className="c0"
        >
          <Link
            replace={false}
            to="/"
          >
            <a
              href="/"
              onClick={[Function]}
            >
              üè†
            </a>
          </Link>
          <span>
            foo
          </span>
        </header>
      </styled.header>
    </Component>
  </Router>
</MemoryRouter>
`;
